git-keeper.py

```diff
@@ -57,32 +66,34 @@ def git_clone_upload(
 ):
     logger.info("Processing repo: %s", repo_url)
     if not repo_url.endswith(".git"):
-        repo_url = repo_url + ".git"
+        repo_url += ".git"
         logger.debug("Appending .git to repo")
-    repo_dir = os.path.join(workdir, os.path.basename(repo_url))
-    repo_tar = repo_dir + ".tar"
+    repo_dir = Path(workdir) / Path(repo_url).name
+    repo_tar = repo_dir.with_suffix(repo_dir.suffix + ".tar")
```

Comment:
This is certainly convoluted just to add a `.tar` suffix to the `repo_dir`. How about 

```suggestion
    repo_tar = Path(str(repo_dir) + ".tar")
```